lib/buffer.js

```diff
@@ -264,7 +264,8 @@ function fromObject(obj) {
   }
 
   if (obj) {
-    if (isArrayBuffer(obj.buffer) || 'length' in obj) {
+    if (isArrayBuffer(obj.buffer) || 'length' in obj ||
+        isSharedArrayBuffer(obj)) {
```

Comment:
@addaleax you're right! I am so sorry about this mistake, I will fix this right away.